entregas/SalmeronJavier/003-ListaSimple/src/listas/basica/List.java

```diff
@@ -0,0 +1,61 @@
+package listas.basica;
+
+public class List {
+    private Node first = null;
+    private int count = 0;
+
+    public int size() {
+        return count;
+    }
+
+    public boolean isEmpty() {
+        return count == 0;
+    }
+
+    public void insert(String value) {
+        System.out.println(" > Llegó " + value);
+        Node newNode = new Node(value);
+        if (this.first == null) {
+            this.first = newNode;
+        } else {
+            Node iterator = this.first;
+            while (iterator.getNext() != null) {
+                iterator = iterator.getNext();
+            }
+            iterator.setNext(newNode);
+        }
+        count++;  // Aumentar la cuenta
+    }
+
+    public void delete() {
+        if (this.first != null) {
+            Node iterator = this.first;
+            Node previous = null;
+            while (iterator.getNext() != null) {
+                previous = iterator;
+                iterator = iterator.getNext();
+            }
+            System.out.println("Se fue " + iterator.getValue());  // Mensaje descriptivo
+            if (previous == null) {
+                this.first = null;
+            } else {
+                previous.setNext(null);
+            }
+            count--;  // Disminuir la cuenta
+        }
+    }
+
+    public String[] listAll() {
+        String[] list = new String[this.size()];
+        Node iterator = this.first;
+        if (iterator == null)
+            return list;
+        int count = 0;
+        while (iterator != null) {
+            list[count] = iterator.getValue();
+            count++;
+            iterator = iterator.getNext();
+        }
+        return list;
+    }
```

Comment:
Aunque muchas veces en clase he manifestado mi personal incomodidad con el this, en estos casos concretos sí me apoyaría en this.count para evitar la ambigüedad al momento de leer el código, porque ambos count hacen referencia a valores en contextos distintos. O bien, cambiaría el nombre de count dentro de listAll.